precompiles/pallet-xcm/XcmInterface.sol

```diff
@@ -0,0 +1,75 @@
+// SPDX-License-Identifier: GPL-3.0-only
+pragma solidity >=0.8.3;
+
+/// @author The Moonbeam Team
+/// @title XCM precompile Interface
+/// @dev The interface that Solidity contracts use to interact with the substrate pallet-xcm.
+interface XCM {
+    // A location is defined by its number of parents and the encoded junctions (interior)
+    struct Location {
+        uint8 parents;
+        bytes[] interior;
+    }
+
+    // Support for Weights V2
+    struct Weight {
+        uint64 refTime;
+        uint64 proofSize;
+    }
+
+    // A way to represent fungible assets in XCM
+    struct AssetLocationInfo {
+        Location location;
+        uint256 amount;
+    }
+
+    struct AssetAddressInfo {
+        address asset;
+        uint256 amount;
+    }
+
+    /// @dev Function to send assets via XCM using transfer_assets() pallet-xcm extrinsic.
+    /// @custom:selector 59df8416
+    /// @param dest The destination chain.
+    /// @param beneficiary The actual account that will receive the tokens in dest.
+    /// @param assets The combination (array) of assets to send.
+    /// @param feeAssetItem The index of the asset that will be used to pay for fees.
+    /// @param weight The weight to be used for the whole XCM operation.
+    /// (uint64::MAX in refTime means Unlimited weight) 
+    function transferAssetsLocation(
+        Location memory dest,
+        Location memory beneficiary,
+        AssetLocationInfo[] memory assets,
+        uint32 feeAssetItem,
+        Weight memory weight
+    ) external;
+
+    /// TODO add docs
+    /// @custom:selector b489262e
+    function transferAssetsToPara20(
+        uint32 paraId,
+        address beneficiary,
```

Comment:
@albertov19 what should be the best type for the `beneficiary` parameter in terms of dev experience? I mean, I need to receive a 20 byte array, but that can be done using the `address` type or the `bytes` type (in the second case I will need to bound the bytes that are being passed to a length of 20 behind the scenes).

Reply:
In this case because it is a `Para20` I think address is perfect.

For `Para32` I think you have to do bytes so

Reply:
Yep, for `Para32` I'm using `bytes32` so no issue there.